File: 219-contains-duplicate-ii/219-contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        seen = {}
        for idx, num in enumerate(nums):
            if idx - seen.get(num, float('-inf')) <= k:
                return True
            seen[num] = idx
        return False
        

File: 219-contains-duplicate-ii/219-contains-duplicate-ii.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_map<int, int> seen;
        
        for(int i = 0; i < nums.size(); ++i){
            int num = nums[i];
            auto val = seen.find(num);
            if (val != seen.end()){
                // cout<< num<< "\n";
                // cout<< i << " "<< val->second << " "<< k << "\n";
                if (i - (val->second) <= k) return true;
            }            
            seen[num] = i;
        }
        for(auto& m : seen){
            cout<< m.first << " " << m.second << "\n";
        }
        return false;        
    }
};